### PR TITLE
Group popup creation code and apply it to more annotation types

### LINKED ISSUE #7014: Highlight/underline/squiggly/strikeout annotations might have popup contents even though there is no Popup entry
https://drive.google.com/uc?export=download&id=0B6YhVF9p-y2AMmlnaURWSjZKNlk is an example of such a document. We already take this into account for Text annotations, but in fact we should do this for all kinds of annotations that have popups.

### PR DESCRIPTION
**Pull request by timvandermeij**
Fixes #7014.

### CODE REVIEW

**Comment by xlc** [comment]
Maybe Highlight/Underline/Squiggly/StrikeOut annotations should have a common base class? Identical code over four classes is not ideal.

**Comment by timvandermeij** [comment]
Now that you brought this up, there are more types that have popups (almost all actually), so I'll update the patch later today to move most popup handling to the Annotation/AnnotationElement base classes. Thanks.

File: src/core/annotation.js [hunk 1]
```diff
@@ -797,8 +797,17 @@ var HighlightAnnotation = (function HighlightAnnotationClosure() {
...
-    this.data.hasPopup = parameters.dict.has('Popup');
+
+    if (!dict.has('C')) {
+      // Fall back to the default background color.
+      this.data.color = null;
+    }
+
+    this.data.hasPopup = dict.has('Popup');
+    this.data.title = stringToPDFString(dict.get('T') || '');
+    this.data.contents = stringToPDFString(dict.get('Contents') || '');
```
Comment (Snuffleupagus) [inline src/core/annotation.js]: So, it appears that this code is repeated six times throughout the file. Would it be possible to re-factor into a helper function instead (similar to the other file)?

Comment (timvandermeij) [inline src/core/annotation.js]: Yes, I've done that in the new commit.

File: src/core/annotation.js [hunk 2]
```diff
@@ -671,15 +689,7 @@ var TextAnnotation = (function TextAnnotationClosure() {
...
-
-    if (!dict.has('C')) {
-      // Fall back to the default background color.
-      this.data.color = null;
-    }
-
-    this.data.hasPopup = dict.has('Popup');
-    this.data.title = stringToPDFString(dict.get('T') || '');
-    this.data.contents = stringToPDFString(dict.get('Contents') || '');
+    this._preparePopup(dict);
```
Comment (Snuffleupagus) [inline src/core/annotation.js]: Nit: I think that you can now remove the variable and simply use `parameters.dict` here.

File: src/core/annotation.js [hunk 3]
```diff
@@ -866,15 +876,7 @@ var FileAttachmentAnnotation = (function FileAttachmentAnnotationClosure() {
...
```
Comment (Snuffleupagus) [inline src/core/annotation.js]: Nit: I think that you can now remove the variable and simply use parameters.dict here.

File: src/core/annotation.js (hunk 2 above)
Comment (timvandermeij) [inline src/core/annotation.js]: Done in the new commit.

File: src/core/annotation.js (hunk 3 above)
Comment (timvandermeij) [inline src/core/annotation.js]: Done in the new commit.

**Comment by Snuffleupagus** [comment]
Looks good, and much better with the helper methods added, thank you for the patch!

/botio makeref

**Pull request merged**